lib/bigstringaf_stubs.c

```diff
@@ -54,7 +54,9 @@ bigstringaf_blit_to_bigstring(value vsrc, value vsrc_off, value vdst, value vdst
     void *src = ((char *)Caml_ba_data_val(vsrc)) + Unsigned_long_val(vsrc_off),
          *dst = ((char *)Caml_ba_data_val(vdst)) + Unsigned_long_val(vdst_off);
     size_t len = Unsigned_long_val(vlen);
+    caml_release_runtime_system();
```

Comment:
This isn't worth the locking overhead.

Reply:
removed runtime release from non-IO code.

Reply:
I don't think the overhead is as big as you might think, but it probably depends on the size of the blit whether it makes sense to release the runtime.

Reply:
In the standard library, OCaml leaves the lock when a `blit` was done on `bigarray`s **and** where the length of the blit is upper or equal to `4096`. I don't know why they did this choice but I suspect something like a possible overhead.

Reply:
If the buffer's memory mapped then in theory the blit could result in a page fault which could make the thread unrunnable. If this happens when the thread has the lock it will block progress of any other threads.

Nobody notices runtime lock overhead until it's a problem. When performed less frequently than every bigstring blit, it amounts to about 5% of execution time. I see it every day.